orchagent/zmqorch.cpp

```diff
@@ -0,0 +1,65 @@
+#include "zmqorch.h"
+
+using namespace swss;
+using namespace std;
+
+extern int gBatchSize;
+
+void ZmqConsumer::execute()
+{
+    SWSS_LOG_ENTER();
+
+    size_t update_size = 0;
+    auto table = static_cast<swss::ZmqConsumerStateTable*>(getSelectable());
+    do
+    {
+        std::deque<KeyOpFieldsValuesTuple> entries;
+        table->pops(entries);
+        update_size = addToSync(entries);
+    } while (update_size != 0);
+
+    drain();
+}
+
+void ZmqConsumer::drain()
+{
+    if (!m_toSync.empty())
+        (static_cast<ZmqOrch*>(m_orch))->doTask(static_cast<ZmqConsumer&>(*this));
```

Comment:
`*this` is already in this class `ZmqConsumer`. Is this casting really needed?

Reply:
Fixed